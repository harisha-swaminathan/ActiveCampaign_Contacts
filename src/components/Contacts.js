import axios from 'axios';
import React from 'react';
import Table from './Table';
import {useState} from 'react';

const Contacts = ()=>{
   const [contacts, setContacts] = useState([]); 
   const [error, setError] = useState('');

    React.useEffect(()=>{
        (async()=>{
            try{
                let allContacts = await axios.get('https://cors-anywhere.herokuapp.com/https://sahmed93846.api-us1.com/api/3/contacts?limit=3',{
                    headers:{
                        'Api-Token':'bcd062dedabcd0f1ac8a568cdcf58660c44d7e79b91763cc1a5d0c03d52c522d851fceb0'
                    }
                });
                allContacts = allContacts.data.contacts;
                setContacts(allContacts);
            }
            catch(error){
                setError(error.message);
            }
            
         })();
    },[]);
    return (
        <div className="table__container">
            <p>{error}</p>
            {contacts.length > 0 &&
                <Table contacts={contacts}/>
            }
        </div>
    );
}

export default Contacts;