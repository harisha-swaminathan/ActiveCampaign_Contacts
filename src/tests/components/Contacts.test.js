import React from 'react';
import {shallow} from 'enzyme';
import contacts from '../fixtures/contacts';
import Contacts from '../../components/Contacts';

let wrapper;
const realUseState = React.useState;

beforeEach(()=>{
    wrapper = shallow(<Contacts/>);
    jest.spyOn(React, "useEffect").mockImplementationOnce(f => f());
    jest.spyOn(React, 'useState').mockImplementationOnce(() => realUseState(contacts.data.contacts))
})
test('should render Contacts correctly', () => {
    expect(wrapper).toMatchSnapshot();
    console.log(wrapper)
});
