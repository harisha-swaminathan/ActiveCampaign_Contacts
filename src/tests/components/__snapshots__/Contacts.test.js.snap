// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Contacts correctly 1`] = `ShallowWrapper {}`;
